### DISCUSSION TITLE
Get Autocomplete Typed text if no result

### DISCUSSION THREAD

**Discussion by elshorbagy** [Megabit/Blazorise#4564] [Q&A] [answered]
Hello,

I'm displaying Cities names in Autocomplete component but if the city does not exist in the database, I want to get the city name from Autocomplete textbox and saved to the database. I tried to get the typed text but I failed, is it possible to do it?
<Autocomplete TItem="City"
                              TValue="int"
                              Data="@_cities"
                              CloseOnSelection
                              TextField="@(( item ) => item.CityName)"
                              ValueField="@(( item ) => item.CityId)"
                              @bind-SelectedValue="@SelectedCityId"
                              @bind-SelectedText="SelectedCityAutoCompleteText"
                              Placeholder="@LanguageService.Keys["City"]"
                              Filter="AutocompleteFilter.StartsWith"
                              FreeTyping
                              CustomFilter="@(( item, searchValue ) => item.CityName.IndexOf(searchValue, 0, StringComparison.CurrentCultureIgnoreCase ) >= 0 )">
                    <NotFoundContent>@LanguageService.Keys["NoResult"]</NotFoundContent>
                </Autocomplete>
  Thanks

**Comment by David-Moreira** [comment, marked as answer]
Hello,
Please refer to our docs
https://blazorise.com/docs/extensions/autocomplete
The bit about Searching with data on demand (ReadData)

Let us know if this is what you are looking for.

**Reply by elshorbagy** [reply]
That helped, thank you.
